app/charts/shared/chart-helpers.tsx

```diff
@@ -287,6 +287,20 @@ export const extractChartConfigComponentIds = ({
   );
 };
 
+export const extractChartConfigUsedCubes = (
+  chartConfig: ChartConfig,
+  { components }: { components: Component[] }
+) => {
+  const componentIds = extractChartConfigComponentIds({
+    chartConfig,
+    includeFilters: false,
+  });
+
+  return componentIds
+    .map((id) => components.find((component) => component.id === id))
+    .filter(truthy); // exclude potential joinBy components
```

Comment:
Do we still need this comment?

Reply:
Here I would also say yes, as I wouldn't know why we need to use `truthy` for – in this case, we have a connection between using it and `joinBy` dimensions. Maybe it would be better to have something like

```typescript
const excludeJoinByComponent = (c: Component) => truthy(c);
```

but I think a comment should suffice here 👍 It was a bit of a pain to catch all the edge-cases of merging of cubes, I'd rather keep more context in places related to them 🥹

Reply:
I think comments that describe a "clear" logic should be generally removed, like this:

```typescript
// Find a component
const component = components.find(c => c.id === id);
```

but when they are related to other, connected parts of the code or generally are not "obvious", it's okay to use them 👍